web_timeline/static/src/js/timeline_controller.js

```diff
@@ -39,7 +39,7 @@ odoo.define('web_timeline.TimelineController', function (require) {
         update: function (params, options) {
             this._super.apply(this, arguments);
```

Comment:
This is probably creating a race condition: `this._super` returns a deferred, which is being ignored and you're creating your own deferred.

Please could you check if this is what you really want here?:

```js
return this._super.apply(this, arguments).then(function () {
   // Your code here
}); //Maybe you need to chain several .then()
```

If your code execution doesn't depend on super's resolution, then you should probably change it for this:

```js
return $.when(this._super.apply(this, arguments), function () {
  // Your code here
});
```